### PR TITLE
Issue #11637: Resolve shellcheck SC2026

### PR DESCRIPTION
**Pull request by shubh220922**
Related to #11637 

https://www.shellcheck.net/wiki/SC2026

### CODE REVIEW

**Comment by shubh220922** [comment]
@nrmancuso @rnveach @romani 

### Testing

#### Testing on master branch

Calling the bash file without removing any target pitest classes.

```
╭─shubh220922@fedora ~/Desktop/checkstyle_project/checkstyle ‹master› 
╰─$ bash .ci/validation.sh check-missing-pitests
List of missing files in pitest profiles: 
╭─shubh220922@fedora ~/Desktop/checkstyle_project/checkstyle ‹master› 
╰─$ 
```

Removing the following pitest target classes from POM.xml to test the same bash file.
```
diff --git a/pom.xml b/pom.xml
index 61acf4172..e57cd8c13 100644
--- a/pom.xml
+++ b/pom.xml
@@ -4289,8 +4289,6 @@
                 <mutator>PRIMITIVE_RETURNS</mutator>
               </mutators>
               <targetClasses>
-                <param>com.puppycrawl.tools.checkstyle.DetailAstImpl*</param>
-                <param>com.puppycrawl.tools.checkstyle.JavadocPropertiesGenerator*</param>
                 <param>com.puppycrawl.tools.checkstyle.XmlLoader*</param>
                 <param>com.puppycrawl.tools.checkstyle.JavaParser*</param>
               </targetClasses>
(END)

```
testing:
```
╭─shubh220922@fedora ~/Desktop/checkstyle_project/checkstyle ‹master●› 
╰─$ bash .ci/validation.sh check-missing-pitests
List of missing files in pitest profiles: src/main/java/com/puppycrawl/tools/checkstyle/DetailAstImpl.java
src/main/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGenerator.java
╭─shubh220922@fedora ~/Desktop/checkstyle_project/checkstyle ‹master●› 
╰─$                                                                                          1 ↵

```
So this is how it behaves in master

#### Testing in my branch (containing the latest push)

Calling the bash file without removing any target pitest classes.

```
╭─shubh220922@fedora ~/Desktop/checkstyle_project/checkstyle ‹SC2026› 
╰─$ bash .ci/validation.sh check-missing-pitests
List of missing files in pitest profiles: 
╭─shubh220922@fedora ~/Desktop/checkstyle_project/checkstyle ‹SC2026› 
╰─$ 
```

Removing the following pitest target classes from POM.xml to test the same bash file.
```
diff --git a/pom.xml b/pom.xml
index 61acf4172..e57cd8c13 100644
--- a/pom.xml
+++ b/pom.xml
@@ -4289,8 +4289,6 @@
                 <mutator>PRIMITIVE_RETURNS</mutator>
               </mutators>
               <targetClasses>
-                <param>com.puppycrawl.tools.checkstyle.DetailAstImpl*</param>
-                <param>com.puppycrawl.tools.checkstyle.JavadocPropertiesGenerator*</param>
                 <param>com.puppycrawl.tools.checkstyle.XmlLoader*</param>
                 <param>com.puppycrawl.tools.checkstyle.JavaParser*</param>
               </targetClasses>
(END)
```
testing:
```
╭─shubh220922@fedora ~/Desktop/checkstyle_project/checkstyle ‹SC2026●› 
╰─$ bash .ci/validation.sh check-missing-pitests
List of missing files in pitest profiles: src/main/java/com/puppycrawl/tools/checkstyle/DetailAstImpl.java
src/main/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGenerator.java
╭─shubh220922@fedora ~/Desktop/checkstyle_project/checkstyle ‹SC2026●› 
╰─$                                                                                          1 ↵
```

File: .ci/validation.sh [hunk 1]
```diff
@@ -42,7 +42,7 @@ check-missing-pitests)
 
   list=($(cat pom.xml | \
     xmlstarlet sel --ps -N pom="http://maven.apache.org/POM/4.0.0" \
-    -t -v '//pom:profile[./pom:id[contains(text(),'pitest')]]//pom:targetClasses/pom:param'))
+    -t -v '//pom:profile[./pom:id[contains(text(),'"'pitest'"')]]//pom:targetClasses/pom:param'))
```
Comment (nrmancuso) [inline .ci/validation.sh:45]: I do not understand why we need all the extra quotes here; why can’t we do https://github.com/checkstyle/checkstyle/pull/12724#discussion_r1102266548 ?

**Review by nrmancuso — CHANGES_REQUESTED** [review]

File: .ci/validation.sh (hunk 1 above)
Comment (nrmancuso) [reply .ci/validation.sh:45]: Shellcheck is not a substitute for humans :) it only sees `'text_in_quotes'text_not_in_quotes'text_in_quotes'` and realizes that the `text_not_in_quotes` is not in the quotes, which is correct. In bash, using `”` allows us to use `'` inside, where another `'` just closes the previous one.

Comment (nrmancuso) [reply .ci/validation.sh:45]: Note that the shellcheck example is a very specific case that doesn’t apply here: in their example, they do not want the `$host` interpolated. This is why they avoided doing what I am suggesting here. We have no variables for interpolation/expansion here, so we should be fine to enclose this expression in double quotes.

Comment (shubh220922) [reply .ci/validation.sh]: Done.

File: .ci/validation.sh [hunk 2]
```diff
@@ -42,7 +42,7 @@ check-missing-pitests)
 
   list=($(cat pom.xml | \
     xmlstarlet sel --ps -N pom="http://maven.apache.org/POM/4.0.0" \
-    -t -v '//pom:profile[./pom:id[contains(text(),'pitest')]]//pom:targetClasses/pom:param'))
+    -t -v "//pom:profile[./pom:id[contains(text(),'pitest')]]//pom:targetClasses/pom:param"))
```
Comment (romani) [inline .ci/validation.sh:45]: we have inner `'pitest'` that is single quoted. I do not think ti will work well

**Review by romani — CHANGES_REQUESTED** [review]
items:

**Review by romani — APPROVED** [review]
sorry I misread code - all good.

I am ok to merge this.

**Review by nrmancuso — APPROVED** [review]

**Pull request merged**